File: backend-ruby/app/models/dang_ky_thi.rb
class DangKyThi < ApplicationRecord
  self.table_name = 'DangKyThi'
  self.primary_key = 'DangKyThiID'
  belongs_to :ky_thi, class_name: "KyThi", foreign_key: "KyThiID"
  belongs_to :sinh_vien, class_name: "SinhVien", foreign_key: "SinhVienID"
  has_one :phan_phong, class_name: "PhanPhong", foreign_key: "DangKyThiID"
  has_one :xep_cho, class_name: "XepCho", foreign_key: "DangKyThiID"
  has_one :diem_danh, class_name: "DiemDanh", foreign_key: "DangKyThiID"
end

File: backend-csharp/Models/DangKyThi.cs
using System.Text.Json.Serialization;

namespace backend_csharp.Models;

public class DangKyThi
{
    public int DangKyThiID { get; set; }
    public int KyThiID { get; set; }
    public int SinhVienID { get; set; }
    public string SoBaoDanh { get; set; } = string.Empty;
    public string TrangThaiDangKy { get; set; } = "registered";
    public DateTime NgayDangKy { get; set; }
    public DateTime TaoLuc { get; set; }
    public DateTime? CapNhatLuc { get; set; }
    public KyThi? KyThi { get; set; }
    public SinhVien? SinhVien { get; set; }
    [JsonIgnore] public PhanPhong? PhanPhong { get; set; }
    [JsonIgnore] public XepCho? XepCho { get; set; }
    [JsonIgnore] public DiemDanh? DiemDanh { get; set; }
}
